iceoryx_posh/include/iceoryx_posh/runtime/service_discovery.hpp

```diff
@@ -38,19 +40,23 @@ class ServiceDiscovery
     /// @param[in] service service string to search for, a nullopt corresponds to a wildcard
     /// @param[in] instance instance string to search for, a nullopt corresponds to a wildcard
     /// @param[in] event event string to search for, a nullopt corresponds to a wildcard
-    /// @return cxx::expected<ServiceContainer, FindServiceError>
+    /// @return ServiceContainer
     /// ServiceContainer: on success, container that is filled with all matching instances
-    /// FindServiceError: if any, encountered during the operation
-    cxx::expected<ServiceContainer, FindServiceError>
-    findService(const cxx::optional<capro::IdString_t>& service,
-                const cxx::optional<capro::IdString_t>& instance,
-                const cxx::optional<capro::IdString_t>& event) noexcept;
-
-    /// @brief requests the serviceRegistryChangeCounter from the shared memory
-    /// @return pointer to the serviceRegistryChangeCounter
-    /// @todo #415 If this method is still used after refactoring, consider to return a reference so that a nullptr
-    /// check is not necessary
-    virtual const std::atomic<uint64_t>* getServiceRegistryChangeCounter() noexcept;
+    ServiceContainer findService(const cxx::optional<capro::IdString_t>& service,
```

Comment:
@FerdinandSpitzschnueffler I slighty had to adapt the API, removing the `expected` as a return value, as this method can't fail anymore.